os/android/iahwc2.cpp

```diff
@@ -801,6 +801,46 @@ HWC2::Error IAHWC2::HwcDisplay::SetColorMode(int32_t mode) {
   return HWC2::Error::None;
 }
 
+HWC2::Error IAHWC2::HwcDisplay::SetColorModeWithRenderIntent(int32_t mode,
+                                                             int32_t intent) {
+  supported(__func__);
+  if (mode < 0)
+    return HWC2::Error::BadParameter;
+  if (intent < 0)
+    return HWC2::Error::BadParameter;
+
+  // TODO: Use the parameter mode to set the color mode for the display to be
+  // used.
+
+  return HWC2::Error::None;
+}
+
+HWC2::Error IAHWC2::HwcDisplay::GetRenderIntents(int32_t mode,
+                                                 uint32_t *outNumIntents,
+                                                 int32_t *outIntents) {
+  supported(__func__);
+  if (mode < 0) {
+    return HWC2::Error::BadParameter;
+  }
+
+  if (outIntents == NULL) {
+    *outNumIntents = getNumRenderIntents();
+    return HWC2::Error::None;
+  }
+
+  *outNumIntents = 0;
+  *outIntents = 0;
+  setNumRenderIntents(*outNumIntents);
+
+  if (mode == HAL_COLOR_MODE_NATIVE) {
```

Comment:
If the mode is not native, the num itents turns back to 0, is it expected?

Reply:
No, from the api description, at least should support HAL_RENDER_INTENT_COLORIMETRIC, i will delete the check condition and recheck again. Thanks!